### PR TITLE
[FLINK-28533] SchemaChange supports updateColumnNullability and updateColumnComment

### PR DESCRIPTION
**Pull request by LadyForest**

### CODE REVIEW

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java [hunk 1]
```diff
@@ -172,34 +175,48 @@ public TableSchema commitChanges(List<SchemaChange> changes) throws Exception {
...
+                    updateColumn(
+                            newFields,
+                            update.fieldName(),
+                            (field) -> {
+                                DataType newType =
+                                        TableSchema.toDataType(
+                                                field.type()
+                                                        .logicalType()
+                                                        .copy(update.newNullability()),
+                                                new AtomicInteger(0));
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java:205]: If the type is a nested type, the fieldId will be wrong.
We can add a `copy(boolean isNullable)` to `DataType`.

**Comment by LadyForest** [comment]
Note: need to rebase master after FLINK-28573 is merged

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/DataType.java [hunk 2]
```diff
@@ -45,6 +45,22 @@ public LogicalType logicalType() {
         return logicalType;
     }
 
+    public static DataType copy(DataType type, boolean isNullable) {
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/DataType.java:48]: maybe a `public abstract DataType copy(boolean isNullable);`?

**Review by JingsongLi — COMMENTED** [review]
Thanks for the update.
Left one minor comment

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**